docs/CIP/CIP-XXXX/README.md

```diff
@@ -0,0 +1,272 @@
+
+Clients connect and authenticate via subscribe messages by providing:
+
+- a digital signature
+- their extended account public key (xpub key)
+
+This two-factor authentication ensures that only authorized clients can access their account data.
+
+The `signature` field is generated by signing with the [extended account private key](https://input-output-hk.github.io/adrestia/static/Ed25519_BIP.pdf) a SHA256 HMAC hashed string. The prehash string is constructed by concatenating the subscribed blockchain(s) + timestamp.
+
+The server verifies the signature and timestamp before serving any data to the client. If any part of the `subscribe` message is invalid, the server sends an error and closes the connection.
+
+#### Transaction Querying
+
+The xpub key also enables efficient querying of client-relevant data:
+
+1. **Address Derivation**
+
+   Using the client's xpub key, our server can deterministically derive a series of public addresses associated with the client's wallet.
+
+2. **Indexing Strategy**
+
+   We index transactions by both payment key hash and stake key hash. This dual-indexing approach allows for comprehensive transaction tracking.
+
+3. **Query Process**
+
+   - The server derives addresses from the client's xpub key
+   - It then queries our index using the corresponding key hashes
+   - This process continues until reaching a predefined [gap limit](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#account-discovery)
+
+> [!NOTE]
+> **Gap Limit:**
+> We stop querying after encountering a specified number of consecutive unused addresses (the gap limit). This prevents unnecessary searches while ensuring we capture all relevant transactions.
+
+### Ordering of Events
+
+Our API needs a reliable measurement of time to construct a stream of ordered, client-relevant, on-chain events that connected clients can easily consume. Time is crucial for clients for various reasons, including:
+
+- monitoring the progress of synchronization up to the chain's tip
+- defining transaction validity intervals
+
+> [!NOTE]
+> Transaction validity intervals allow clients to submit transactions with a specified lifetime, which serves several important purposes:
+>
+> - supports operations that need to occur within specific timeframes
+> - helps manage network congestion by allowing expired transactions to be discarded
+> - protects against replay attacks by ensuring transactions can't be resubmitted after expiration
+
+We borrow the term `point` from the Cardano blockchain to introduce a sever-sent message partial. For instance, if a new transaction is deemed relevant for a connected client, the server enriches the message by adding the current "`point` in on-chain time" of the respective chain. For Cardano, this typically looks as follows:
+
+#### Cardano Point
+
+```json
+{
+  "point": {
+    "slot": 127838345,
+    "hash": "9f06ab6ecce25041b3a55db6c1abe225a65d46f7ff9237a8287a78925b86d10e"
+  }
+  /* ... */
+}
+```
```

Comment:
The _point_  serves a particular purpose, which is to capture the current "time". Anything outside of this would be considered unnecessary and we would be bleeding into this concept being more of a _block header_